input-output-hk/lace
# feat(staking): LW-8179 Make CTAs in staking management drawer dynamic

# Checklist

- [x] JIRA - LW-8179

---

## Proposed solution

Extended staking management drawer state machine footer logic to dynamically change the footer depending on the portfolio state.

Moreover, this PR refactors the `footerVisible` variable to include the logic within the Footers mapping which simplifies reasoning about the visibility of the footer as it moves the logic closer to the actually impacted Flow/Step/Element

## Testing

1. Rebalance the portfolio and check that upon clicking on "Manage" no footer is visible. Check that upon removing a pool "Confirm staking" button appears
2. Drift the portfolio e.g. by sending a large portion of funds to yourself (provided dekegating at least 2 pools) and check that upon clicking "Manage" the button text is "Rebalance staking"
3. Select manually some pools in "Browse pools" section, click "Next" and check that the management drawer that appears has below a button with the text "Confirm staking"
4. Check that the buttons mentioned above work as intended
5. Check that button texts match Figma designs from the task description

## Screenshots

before, all management drawer initial screens looked the same:

![Screenshot 2023-09-27 at 12 39 01](https://github.com/input-output-hk/lace/assets/4980147/4b2713af-34b0-45d4-8682-42323a4c10de)

after:

no drift, no editation of current portfolio:
![Screenshot 2023-09-27 at 12 31 33](https://github.com/input-output-hk/lace/assets/4980147/fa9e8c16-8a8a-4904-aab0-434aa08e1a4f)

edited portfolio/new selection:
![Screenshot 2023-09-27 at 12 31 53](https://github.com/input-output-hk/lace/assets/4980147/4ec08e98-49e2-4236-806b-b91cc1a4e7fb)

current portfolio drifted:
![Screenshot 2023-09-27 at 12 32 38](https://github.com/input-output-hk/lace/assets/4980147/bdab1767-768a-46ce-b781-b2502da65b94)




---
# Allure report
`allure-report-publisher` generated test report!



**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/2251/6338907738/index.html) for [b8038f2b](https://github.com/input-output-hk/lace/pull/575/commits/b8038f2b042024e4d84027cb6e27668f6c769429)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 31     | 1      | 0       | 0     | 32    | ❌     |
